pandas/tests/arrays/test_integer.py

```diff
@@ -750,9 +755,11 @@ def test_reduce_to_float(op):
 def test_astype_nansafe():
     # see gh-22343
     arr = integer_array([np.nan, 1, 2], dtype="Int8")
-    msg = "cannot convert float NaN to integer"
+    # XXX: determine the proper exception here, from int(NA).
```

Comment:
Should `int(NA)` be a TypeError or ValueError?

Reply:
`int(NA)` being a TypeError is the default that you get from Python (I suppose we could change this through `__index__`), but to me that seems fine to keep. 

In the astype, however, we can in principle have control over which error gets raised? 
I see that in Boolean.astype I explicitly check if the dtype is integer-like, and then raise a ValueError with a custom message if there are NAs present.